### PR TITLE
Feature/selection box

### PR DESCRIPTION
**Pull request by owenxr**
For Selection box
- Shows the current words formed by the character selected by cursor
- Shows current letter that is highlighted by cursor 
- Add `get_words_at` and `get_words_with_axis` in `board.ml` 
- Implement `draw_selection` and `draw_sel_letter` in `main.ml` for drawing selection box features 

Screen Shot 2021-11-29 at 2 10 35 PM
.

### CODE REVIEW

**Comment by owenxr** [comment]
Screen Shot 2021-11-29 at 2 31 11 PM

**Review by isabellawu90 — APPROVED** [review]

File: lib/board.ml [hunk 1]
```diff
@@ -159,14 +159,83 @@ let get_words_impl
...
+        (find_word position Horizontal, Horizontal);
+      ]
+
+(** [get_words_at board position] returns words that are found by
+    starting at [position] and searching outwards. *)
+let get_words_at (board : board) (position : position) : string list =
+  let lst = get_words_with_axis board position in
+  List.map (fun x -> fst x) lst
+
+(* let get_words_at (board : board) (position : position) : string list
```
Comment (laptou) [inline lib/board.ml:219]: Since this code is not being used anymore, it should be deleted

**Review by kyralee22 — APPROVED** [review]
This is fancy lol looks cool

File: lib/board.ml (hunk 1 above)
Comment (owenxr) [reply lib/board.ml]: Fixed

**Review by laptou — APPROVED** [review]
lgtm

**Pull request merged**